geo/src/algorithm/centroid.rs

```diff
@@ -340,28 +339,136 @@ where
     type Output = Option<Point<T>>;
 
     fn centroid(&self) -> Self::Output {
-        if self.0.is_empty() {
-            return None;
+        centroid_of_coords(self.iter().map(|p| p.0)).map(Point)
+    }
+}
+
+impl<T> Centroid for Geometry<T>
+where
+    T: CoordFloat + FromPrimitive + Sum + Default,
+{
+    type Output = Option<Point<T>>;
+
+    crate::geometry_delegate_impl! {
+        fn centroid(&self) -> Self::Output;
+    }
+}
+
+impl<T> Centroid for GeometryCollection<T>
+where
+    T: CoordFloat + FromPrimitive + Sum + Default,
+{
+    type Output = Option<Point<T>>;
+    fn centroid(&self) -> Self::Output {
+        use crate::algorithm::dimensions::{Dimensions, HasDimensions};
+
```

Comment:
Can we add a comment here explaining the contents of the tuple? Something like:

```
// (Option<T>, T) represents (centroid, weight)
```

or

```
// (centroid: Option<T>, weight: T)
```

Reply:
Good call - I added a doc for the tuple as well as an overview of the algorithm, because I think it's not obvious.